src/core/src/sketch/minhash.rs

```diff
@@ -460,30 +460,12 @@ impl KmerMinHash {
 
     pub fn count_common(&self, other: &KmerMinHash, downsample: bool) -> Result<u64, Error> {
         if downsample && self.max_hash != other.max_hash {
-            if self.max_hash < other.max_hash {
-                let mut new_mh = KmerMinHash::new(
-                    other.num,
-                    other.ksize,
-                    other.hash_function,
-                    other.seed,
-                    self.max_hash,
-                    other.abunds.is_some(),
-                );
-                new_mh.add_many(&other.mins)?;
-                self.count_common(&new_mh, false)
-            } else {
-                // other.max_hash < self.max_hash
-                let mut new_mh = KmerMinHash::new(
-                    self.num,
-                    self.ksize,
-                    self.hash_function,
-                    self.seed,
-                    other.max_hash,
-                    self.abunds.is_some(),
-                );
-                new_mh.add_many(&self.mins)?;
-                new_mh.count_common(other, false)
-            }
+            let cmp = self.max_hash < other.max_hash;
+            let a = if cmp { self } else { other };
+            let b = if cmp { other } else { self };
```

Comment:
I think this is better, but still a bit confusing... How about
```rust
let (a, b) = if self.max_hash < other.max_hash {                                                                                                                                
  (self, other)                                                                                                                                                               
} else {                                                                                                                                                                        
  (other, self)                                                                                                                                                               
};  
```
(`a` and `b` are also not very good variable names, but not sure what to suggest =])

Reply:
oooh tuple unpacking!!